app/bundles/LeadBundle/EventListener/PointSubscriber.php

```diff
@@ -51,18 +65,60 @@ public function onTriggerBuild(TriggerBuilderEvent $event)
                 'eventName' => PointEvents::TRIGGER_ON_EVENT_EXECUTE,
             ]
         );
+
+        $event->addEvent(
+            'lead.changestage',
+            [
+                'group'     => 'mautic.lead.point.trigger',
+                'label'     => 'mautic.lead.lead.events.changestage',
+                'formType'  => StageActionChangeType::class,
+                'eventName' => PointEvents::TRIGGER_ON_EVENT_EXECUTE,
+            ]
+        );
     }
 
     public function onTriggerExecute(TriggerExecutedEvent $event): void
     {
-        if ('lead.changetags' !== $event->getTriggerEvent()->getType()) {
-            return;
+        if ('lead.changetags' === $event->getTriggerEvent()->getType()) {
+            $this->handelChangeTags($event);
+        } elseif ('lead.changestage' === $event->getTriggerEvent()->getType()) {
+            $this->handelChangeStage($event);
         }
+    }
 
-        $properties = $event->getTriggerEvent()->getProperties();
-        $addTags    = $properties['add_tags'] ?: [];
-        $removeTags = $properties['remove_tags'] ?: [];
+    /**
+     * Add or remove tags from a contact based on the trigger event.
+     */
+    private function handelChangeTags(TriggerExecutedEvent $event): void
+    {
+        $properties         = $event->getTriggerEvent()->getProperties();
+        $addTags            = $properties['add_tags'] ?: [];
+        $removeTags         = $properties['remove_tags'] ?: [];
 
         $this->leadModel->modifyTags($event->getLead(), $addTags, $removeTags);
     }
+
+    /**
+     * Change the stage of a contact based on the trigger event.
+     *
+     * @todo allow to remove a stage
+     */
+    private function handelChangeStage(TriggerExecutedEvent $event): void
+    {
+        $stageId  = (int) $event->getTriggerEvent()->getProperties()['stage'];
+        $stage    = $this->stageHelper->getStage($stageId);
+
+        if (null === $stage) {
+            throw new \InvalidArgumentException("Stage for ID $stageId not found");
```

Comment:
I'm 🤔 if a stage is deleted but still referenced in a point subscriber, what will happen? Will this cause 500 errors on all requests that would trigger this stage change?

Reply:
I was thinking this is a good idea. The calling code will then have to handle the exception in the appropriate way. 
The other option is to at least add a dependency for the logger. But this seamed a bit much for this one case? What do you think?

Reply:
yes, I agree. Since it is a Trigger I thought it is handled somewhere. So it is acceptable as the user will not see it.